build-tools/packages/build-cli/src/library/packageExports.ts

```diff
@@ -248,68 +248,46 @@ export function queryTypesResolutionPathsFromPackageExports<TOutKey>(
 }
 
 /**
- * Finds the path to the types of a package using the package's export map or types/typings field.
- * If the path is found, it is returned. Otherwise it returns undefined.
+ * Finds the path of an API level from a package's export map.
+ * The function uses the specified conditions to locate the appropriate path.
+ * If no conditions are provided, it defaults to using the `default` condition.
  *
- * This implementation uses resolve.exports to resolve the path to types for a level.
- *
- * @param packageJson - The package.json object to check for types paths.
- * @param level - An API level to get types paths for.
- * @returns A package relative path to the types.
- *
- * @remarks
- *
- * This implementation loosely follows TypeScript's process for finding types as described at
- * {@link https://www.typescriptlang.org/docs/handbook/modules/reference.html#packagejson-main-and-types}. If an export
- * map is found, the `types` and `typings` field are ignored. If an export map is not found, then the `types`/`typings`
- * fields will be used as a fallback _only_ for the public API level (which corresponds to the default export).
- *
- * Importantly, this code _does not_ implement falling back to the `main` field when `types` and `typings` are missing,
- * nor does it look up types from DefinitelyTyped (i.e. \@types/* packages). This fallback logic is not needed for our
- * packages because we always specify types explicitly in the types field, and types are always included in our packages
- * (as opposed to a separate \@types package).
+ * @param packageJson - The package.json object to check for paths.
+ * @param level - An API level to get paths for.
+ * @param conditions - Export conditions to apply. Fallback to `default` if no condition is passed
+ * @returns A package relative path.
  */
-export function getTypesPathFromPackage(
+export function getExportPathFromPackage(
```

Comment:
Use of this seems incorrect (fallible). When I looked at this a while back it didn't feel familiar, and something felt off. We can get away with using this for node10 and probably for type tests. Node10 has narrow "conditions". Typetests are also narrowly focused on an entrypoint and just types.
Some packages have conditions that are beyond "default" and "types". That is why the prior version of things does not use `resolve.exports` package which requires you to know the conditions. Instead, we want to find all of the paths under an export given the generation criteria and get their conditions. It was `queryTypesResolutionPathsFromPackageExports` that I was expecting a generic form of or to be adapted away from strictly for "types" conditions. (Would have matched the original command.)
Probably this pattern is why some other cases seem hard to account for - like stopping automatic generation of a file temporarily.
I need to think about this.